File: shopfloor_stock_picking_partner_note/services/checkout.py
# Copyright 2024 Camptocamp SA (https://www.camptocamp.com)
# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
import os

from odoo.addons.shopfloor.services.checkout import Checkout


class CheckoutExt(Checkout):
    _inherit = "base.shopfloor.process"

    def _data_for_packing_info(self, picking):
        res = super()._data_for_packing_info(picking)
        note = picking.note
        if note:
            res += "{}{}".format(os.linesep, note).strip()
        return res

File: shopfloor_stock_picking_partner_note/static/src/scenario/checkout.js
/**
 * Copyright 2024 Camptocamp SA (http://www.camptocamp.com)
 * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 */

import {process_registry} from "/shopfloor_mobile_base/static/wms/src/services/process_registry.js";

// stock_picking_partner_note makes the 'note' field in pickings an HTML field,
// which means that we need to display the note as HTML in the shopfloor frontend as well.
const checkout_scenario = process_registry.get("checkout");
const template = checkout_scenario.component.template;
const new_template = template.replace(
    'v-text="state.data.picking.note"',
    'v-html="state.data.picking.note"'
);

const CheckoutStockPickingPartnerNote = process_registry.extend("checkout", {
    template: new_template,
});

process_registry.replace("checkout", CheckoutStockPickingPartnerNote);
